py/dml/types.py

```diff
@@ -1158,14 +1181,15 @@ def clone(self):
                          self.const)
 
     def declaration(self, var):
-        params = [ t.declaration("arg%d" % n)
-                   for (n, t) in enumerate(self.input_types) ]
+        arglist = ', '.join(t.declaration("arg%d" % n)
+                            for (n, t) in enumerate(self.input_types))
         if self.varargs:
-            varargs = ", ..."
+            arglist += ", ..."
+        if self.const:
+            funtype = self.output_type.declaration(f'({arglist})')
+            return f'const typeof({funtype}) {var}'
```

Comment:
I don't (possibly because someone thought about it and realized there is no risk).
However, you do get a hard error for
```
        typedef const typeof(void (int)) *t1;
        typedef       typeof(void (int)) *t2;
        void (*f)(t1) = NULL;
        void (*g)(t2) = f;
```
... because pointer incompatibility means invalid C.

Reply:
Reworked now.